File: seed/php-sdk/grpc-proto-exhaustive/src/Types/ListResponse.php
<?php

namespace Seed\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;
use Seed\Core\Types\ArrayType;

class ListResponse extends JsonSerializableType
{
    /**
     * @var ?array<ListElement> $columns
     */
    #[JsonProperty('columns'), ArrayType([ListElement::class])]
    public ?array $columns;

    /**
     * @var ?Pagination $pagination
     */
    #[JsonProperty('pagination')]
    public ?Pagination $pagination;

    /**
     * @var ?string $namespace
     */
    #[JsonProperty('namespace')]
    public ?string $namespace;

    /**
     * @var ?Usage $usage
     */
    #[JsonProperty('usage')]
    public ?Usage $usage;

    /**
     * @param array{
     *   columns?: ?array<ListElement>,
     *   pagination?: ?Pagination,
     *   namespace?: ?string,
     *   usage?: ?Usage,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->columns = $values['columns'] ?? null;
        $this->pagination = $values['pagination'] ?? null;
        $this->namespace = $values['namespace'] ?? null;
        $this->usage = $values['usage'] ?? null;
    }
}

File: seed/pydantic-v2/grpc-proto-exhaustive/types/list_response.py
from typing import List, Optional

from .types.list_element import ListElement
from .types.pagination import Pagination
from .types.usage import Usage
from core.datetime_utils import serialize_datetime
from dt import datetime

from pydantic import BaseModel


class ListResponse(BaseModel):
    columns: Optional[List[ListElement]] = None
    pagination: Optional[Pagination] = None
    namespace: Optional[str] = None
    usage: Optional[Usage] = None

    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}
